contribs/freight/src/main/java/org/matsim/freight/carriers/analysis/CarrierPlanAnalysis.java

```diff
@@ -54,103 +55,149 @@ public class CarrierPlanAnalysis {
 	private static final Logger log = LogManager.getLogger(CarrierPlanAnalysis.class);
 	public final String delimiter;
 
+	public enum CarrierAnalysisType {onlyDemand, completeAnalysis}
+
 	final Carriers carriers;
 
 	public CarrierPlanAnalysis(String delimiter, Carriers carriers) {
 		this.delimiter = delimiter;
 		this.carriers = carriers;
 	}
 
-	public void runAnalysisAndWriteStats(String analysisOutputDirectory) throws IOException {
+	public void runAnalysisAndWriteStats(String analysisOutputDirectory, CarrierAnalysisType analysisType) {
 		log.info("Writing out carrier analysis ...");
-		//Load per vehicle
-		String fileName = Path.of(analysisOutputDirectory).resolve("Carrier_stats.tsv").toString();
 
+		Path path = Path.of(analysisOutputDirectory);
+		String fileName = switch (analysisType) {
+			case onlyDemand -> path.resolve("CarriersPlan_stats_demand.tsv").toString();
+			case completeAnalysis -> path.resolve("CarriersPlan_stats_solution.tsv").toString();
+		};
 		try (BufferedWriter bw1 = new BufferedWriter(new FileWriter(fileName))) {
+			String headerGeneral = String.join(delimiter,
+				"carrierId",
+				"nuOfJspritIterations",
+				"nuOfPossibleVehicleTypes",
+				"nuOfPossibleVehicles",
+				"nuOfServiceLocations(input)",
```

Comment:
are "(" and ")" good characters if one would like to to further analysis e.g. in R? 
(this is valid for all headers)